src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -4,35 +4,108 @@
 import io.javaoperatorsdk.operator.api.reconciler.Context;
 import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
 import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
 import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
 import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
 import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
 import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.kafka.access.model.BindingStatus;
 import io.strimzi.kafka.access.model.KafkaAccess;
+import io.strimzi.kafka.access.model.KafkaAccessStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.Collections;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 @ControllerConfiguration
 public class KafkaAccessReconciler implements Reconciler<KafkaAccess>, EventSourceInitializer<KafkaAccess> {
 
+    private static final String TYPE_SECRET_KEY = "type";
+    private static final String TYPE_SECRET_VALUE = "kafka";
+    private static final String PROVIDER_SECRET_KEY = "provider";
+    private static final String PROVIDER_SECRET_VALUE = "strimzi";
+    private static final String SECRET_TYPE = "servicebinding.io/kafka";
+
     private final KubernetesClient kubernetesClient;
+    private final Map<String, String> commonSecretData = new HashMap<>();
+    private final Map<String, String> commonSecretLabels = new HashMap<>();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAccessOperator.class);
 
-    public KafkaAccessReconciler(KubernetesClient kubernetesClient) {
+    public KafkaAccessReconciler(final KubernetesClient kubernetesClient) {
         this.kubernetesClient = kubernetesClient;
+        final Base64.Encoder encoder = Base64.getEncoder();
+        commonSecretData.put(TYPE_SECRET_KEY, encoder.encodeToString(TYPE_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretData.put(PROVIDER_SECRET_KEY, encoder.encodeToString(PROVIDER_SECRET_VALUE.getBytes(StandardCharsets.UTF_8)));
+        commonSecretLabels.put(Utils.MANAGED_BY_LABEL_KEY, Utils.KAFKA_ACCESS_LABEL_VALUE);
     }
 
     @Override
-    public UpdateControl<KafkaAccess> reconcile(KafkaAccess kafkaAccess, Context context) {
-        return UpdateControl.noUpdate();
+    public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final Context context) {
+        final String kafkaAccessName = kafkaAccess.getMetadata().getName();
+        LOGGER.info("Reconciling KafkaAccess " + kafkaAccessName);
+        final Map<String, String> data  = new HashMap<>(commonSecretData);
+        kubernetesClient
```

Comment:
Thanks @csviri, that's a good point. I guess at the moment this is quite a dumb operator that just repeatedly creates the secret, rather than checking if it needs updating. When you say there is one in the cache, does that mean I can just check the context to pick out what the secret is currently set to, without having to actually query K8s to get the current content of the secret for the comparison?

Reply:
Informer maintains a local in memory cache regarding the watched objects, so instead of call the API you can do something like this in this example:
https://github.com/java-operator-sdk/java-operator-sdk/blob/c43965e491bd3a092ea0fc1e2f243bb6cc4218de/sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/TomcatReconciler.java#L61

In your case would be ` context.getSecondaryResource(Secret.class)`, If the Secret is not in the cache it means it's not created yet.

An alternative way is just to hold a reference in the operator to the informer, it has a nice api to do all kind of things with the cache.